File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minn=prices[0],profit=0;
        int cost=0;
        for(int i=1;i<prices.size();i++){
            cost=prices[i]-minn;
            profit=max(profit,cost);
            minn=min(minn,prices[i]);
            
        }
    return profit;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit=0

        mini=prices[0]

        maxi=0

        for i in range(len(prices)):

            if prices[i] < mini:
                mini=prices[i]

            profit=prices[i]-mini
            maxi=max(maxi,profit)

        return maxi